src/object/C4Movement.cpp

```diff
@@ -275,31 +341,36 @@ void C4Object::DoMovement()
 		{
 			// Next step
 			int step = Sign(new_y - fix_y);
-			if ((iContact=ContactCheck(GetX(), GetY() + step, nullptr, ydir > 0)))
+			if ((iContact = ContactCheck(GetX(), GetY() + step, nullptr, ydir > 0)))
 			{
-				fAnyContact=true; iContacts |= t_contact;
+				fAnyContact = true;
+				iContacts |= t_contact;
 				new_y = fix_y;
 				// Vertical contact horizontal friction
-				ApplyFriction(xdir,ContactVtxFriction(this));
+				ApplyFriction(xdir, ContactVtxFriction(this));
 				// Redirection slide or rotate
-				if (!ContactVtxCNAT(this,CNAT_Left))
-					RedirectForce(ydir,xdir,-1);
-				else if (!ContactVtxCNAT(this,CNAT_Right))
-					RedirectForce(ydir,xdir,+1);
+				if (!ContactVtxCNAT(this, CNAT_Left))
+				{
+					RedirectForce(ydir, xdir, -1);
+				}
+				else if (!ContactVtxCNAT(this, CNAT_Right))
+				{
+					RedirectForce(ydir, xdir, +1);
+				}
 				else
 				{
 					// living things are always capable of keeping their rotation
-					if (OCF & OCF_Rotate) if (iContact==1) if (!Alive)
-							{
-								RedirectForce(ydir,rdir,-ContactVtxWeight(this));
-								fRedirectYR=1;
-							}
-					ydir=0;
+					if (OCF & OCF_Rotate) if (iContact == 1) if (!Alive)
```

Comment:
```suggestion
					if ((OCF & OCF_Rotate) && iContact == 1 && !Alive)
```

Reply:
```suggestion
					if ((OCF & OCF_Rotate) && !Alive && iContact == 1)
```
Did not really understand how suggestion works. Anyway, I think that the order can be changed, maybe even have !Alive first.

Reply:
What about equality checks inverted? 1 == iContact

Reply:
That reads like Yoda and isn't really useful because compilers like to complain about accidental assignments.

Reply:
Funny though, because we had several people at work who thought that this is super safe for avoiding accidental assignments (well, it is) and the thing to always do.

Reply:
It is safe(r), but clang warns about it by default, gcc does it as part of `-Wall` (or explicit by `-Wparentheses`, which works for clang too), and for MSVC you need to enable it explicitly with `/we4706`.